File: app/src/main/java/com/absinthe/kage/device/cmd/InquiryPlayerStatusCommand.java
package com.absinthe.kage.device.cmd;

import com.absinthe.kage.connect.protocol.IpMessageConst;
import com.absinthe.kage.device.Command;
import com.absinthe.kage.device.CommandBuilder;
import com.absinthe.kage.device.Device;
import com.absinthe.kage.device.client.Client;
import com.absinthe.kage.media.audio.AudioPlayer;

import java.io.IOException;

public class InquiryPlayerStatusCommand extends Command {

    public InquiryPlayerStatusCommand() {
        cmd = IpMessageConst.MEDIA_GET_PLAYER_STATUS;
    }

    @Override
    public String pack() {
        return new CommandBuilder()
                .with(this)
                .build();
    }

    @Override
    public void doWork(Client client, String received) {
        SetPlayStatusCommand command = new SetPlayStatusCommand();
        command.statusCode = AudioPlayer.getInstance(client.getContext()).getPlayState();

        try {
            client.writeToStream(command.pack());
        } catch (IOException e) {
            e.printStackTrace();
            client.offline();
        }
    }

    @Override
    public boolean parseReceived(String received) {
        return true;
    }
}

File: app/src/main/kotlin/com/absinthe/kage/device/cmd/InquiryPlayerStatusCommand.kt
package com.absinthe.kage.device.cmd

import com.absinthe.kage.connect.protocol.IpMessageConst
import com.absinthe.kage.connect.proxy.BaseProxy
import com.absinthe.kage.connect.proxy.MODE_AUDIO
import com.absinthe.kage.device.Command
import com.absinthe.kage.device.CommandBuilder
import com.absinthe.kage.device.client.Client
import com.absinthe.kage.media.audio.AudioPlayer
import com.absinthe.kage.media.video.LocalVideoPlayback
import java.io.IOException

class InquiryPlayerStatusCommand : Command() {

    override fun pack(): String {
        return CommandBuilder()
                .with(this)
                .build()
    }

    override fun doWork(client: Client, received: String) {
        val command = SetPlayStatusCommand().apply {
            statusCode = if (BaseProxy.CURRENT_MODE == MODE_AUDIO)
                AudioPlayer.playState
            else
                LocalVideoPlayback.INSTANCE?.state ?: 0
        }

        try {
            client.writeToStream(command.pack())
        } catch (e: IOException) {
            e.printStackTrace()
            client.offline()
        }
    }

    override fun parseReceived(received: String): Boolean {
        return true
    }

    init {
        cmd = IpMessageConst.MEDIA_GET_PLAYER_STATUS
    }
}